app/views/configure/queries.phtml

```diff
@@ -71,5 +71,5 @@
 		}
 	?>
 	</div>
-
+	<script src="../scripts/slider.js?<?= @filemtime(PUBLIC_PATH . '/scripts/slider.js') ?>"></script>
```

Comment:
If you add the script the proper way (i.e., through the controller) I assume that will also take care of your order issue.

Reply:
grep for appendScript

Reply:
the issue is the download time of main.js and slider.js
sometimes main.js is faster loaded than slider.js, sometimes otherway around.


This should solve the issue because it adds init_slider_afterDOM() after DOMContentLoaded event. But it does not work in a propper way.
```
if (document.readyState && document.readyState !== 'loading') {
	init_slider_afterDOM();
} else {
	if (window.console) {
		console.log('FreshRSS slider waiting for DOMContentLoaded…');
	}
	document.addEventListener('DOMContentLoaded', init_slider_afterDOM, false);
}
```

Reply:
Now the slider code is included into main.js. It will solve the issue best.

Reply:
I'm fine with that, but I'm not convinced there's a good reason for async here. If you simply defer it they're executed *in order* right before DOMContentLoaded.

Reply:
I tried to use "defer" but it did not solve the issue :/

Reply:
In what sense? I mean this:
```diff
diff --git a/app/FreshRSS.php b/app/FreshRSS.php
index dc5ed099..b395a5d1 100644
--- a/app/FreshRSS.php
+++ b/app/FreshRSS.php
@@ -129,7 +129,7 @@ class FreshRSS extends Minz_FrontController {
                if (Minz_Request::controllerName() !== 'index') {
                        FreshRSS_View::prependScript(Minz_Url::display('/scripts/extra.js?' . @filemtime(PUBLIC_PATH . '/scripts/extra.js')));
                }
-               FreshRSS_View::prependScript(Minz_Url::display('/scripts/main.js?' . @filemtime(PUBLIC_PATH . '/scripts/main.js')));
+               FreshRSS_View::prependScript(Minz_Url::display('/scripts/main.js?' . @filemtime(PUBLIC_PATH . '/scripts/main.js')), true, false);
        }
 
        private static function loadNotifications() {
```

Reply:
Order will be ensured by the browser.

Reply:
Anyway, it's a moot point. I thought by slider.js you were referring to a pre-existing file.

Reply:
Sorry, my fault. I created slider.js as a new file.